### PR TITLE
Configuration option for changing the log level

### PR DESCRIPTION
**Pull request by rabidaudio**
This adds a configuration option `logLevel` which can be one of the [Gradle log levels](https://docs.gradle.org/current/userguide/logging.html#sec:choosing_a_log_level).

I realize it's a bit strange to log test results on any level other than `lifecycle` but the use case we're experiencing is that Gradle is printing a lot of noise for various compilation tasks (an Android project with multiple modules) and the test results get lost in the noise, so we are going to pass `--quiet` and have our test results log on `quiet`. Totally understand if you don't want to support this use case, we can keep using our fork. 

I added a unit test and functional test, but I forgot to update the docs. If you're interested in accepting this PR let me know and I'll update the docs.

### CODE REVIEW

**Comment by radarsh** [comment]
Hi, @rabidaudio, thanks for submitting this PR. I will review this in the near future and provide feedback.

**Comment by mattdkerr** [comment]
looks good to me-- has this been tested in kotlin-dsl?

File: src/test/groovy/com/adarshr/gradle/testlogger/logger/ConsoleLoggerSpec.groovy [hunk 1]
```diff
@@ -40,4 +40,14 @@ class ConsoleLoggerSpec extends Specification {
...
     }
+
+    def "configurable log level"() {
+        given:
+            consoleLogger = new ConsoleLogger(loggerMock, DEBUG, textRendererMock)
+            textRendererMock.render('text to be logged') >> 'rendered ansi text'
+        when:
+        consoleLogger.log('text to be logged')
+        then:
+        1 * loggerMock.log(DEBUG, 'rendered ansi text')
```
Comment (radarsh) [inline src/test/groovy/com/adarshr/gradle/testlogger/logger/ConsoleLoggerSpec.groovy]: Lines 49 and 51 should be indented according to the convention used for Spock tests in the rest of this project.

File: src/main/groovy/com/adarshr/gradle/testlogger/TestLoggerExtension.groovy [hunk 2]
```diff
@@ -16,6 +17,7 @@ import static org.gradle.api.tasks.testing.logging.TestExceptionFormat.FULL
 class TestLoggerExtension {
 
     ThemeType theme = STANDARD
+    LogLevel logLevel = LogLevel.LIFECYCLE
```
Comment (radarsh) [inline src/main/groovy/com/adarshr/gradle/testlogger/TestLoggerExtension.groovy]: We should also document this new property in the README under the relevant sections.

**Comment by rabidaudio** [comment]
Sorry for the delay here. I addressed the issues and rebased against develop.

**Pull request merged**